src/main/java/com/laytonsmith/tools/langserv/LangServ.java

```diff
@@ -727,16 +748,38 @@ public void doCompilation(CompletableFuture<ParseTree> future, Executor threadPo
 				compilerEnv.setLogCompilerWarnings(false); // No one would see them
 				GlobalEnv gEnv = env.getEnv(GlobalEnv.class);
 
-				URI uuri = new URI(uri);
+
 				File f;
-				if("untitled".equals(uuri.getScheme())) {
-					// For open files that aren't saved to disk, the client sends something like "untitled:untitled-1",
-					// which isn't a valid file provider, so we can't call Paths.get on it. Instead, we just mock
-					// the name here. We also need to provide getAbsoluteFile, so that the below call to getParentFile
-					// will work.
-					f = new File(uuri.getSchemeSpecificPart()).getAbsoluteFile();
-				} else {
-					f = Paths.get(uuri).toFile();
+				{
+					URI uuri = new URI(uri);
+					if("untitled".equals(uuri.getScheme())) {
+						// For open files that aren't saved to disk, the client sends something like "untitled:untitled-1",
+						// which isn't a valid file provider, so we can't call Paths.get on it. Instead, we just mock
+						// the name here. We also need to provide getAbsoluteFile, so that the below call to getParentFile
+						// will work.
+						f = new File(uuri.getSchemeSpecificPart()).getAbsoluteFile();
+					} else {
+						f = Paths.get(uuri).toFile();
+					}
+				}
+
+				// SA is currently not async safe, so we just manually synch outside for now. This is obviously
+				// bad and should be fixed.
+				synchronized(LangServ.class) {
+					Set<File> autoIncludes = new HashSet<>();
+					for(WorkspaceFolder folder : workspaceFolders) {
+						URI uuri = new URI(folder.getUri());
+						File ai = Paths.get(uuri).toFile();
+						FileUtil.recursiveFind(ai, (r) -> {
+							if(r.isFile() && r.getAbsolutePath().endsWith("auto_include.ms")) {
+								autoIncludes.add(r);
+							}
+						});
```

Comment:
Don't forget that it needs to ignore .disabled and .library directories here.

Reply:
Ah, crap, yeah.